pkg/streamingpromql/operators/binary_operation.go

```diff
@@ -812,7 +822,13 @@ func (b *BinaryOperation) computeResult(left types.InstantVectorSeriesData, righ
 			// Timestamps match at this step
 			resultFloat, resultHist, ok, err := b.opFunc(lF, rF, lH, rH)
 			if err != nil {
-				return types.InstantVectorSeriesData{}, err
+				err = functions.NativeHistogramErrorToAnnotation(err, b.emitAnnotation)
+				if err == nil {
+					// Error was converted to an annotation, continue without emitting a sample here.
+					ok = false
```

Comment:
(nit) I think `ok` will always be false when there is an error. No harm in being explicit though.

Reply:
Yep, that's exactly my thinking.